### PR TITLE
Update equinix sdk

### LINKED ISSUE #265: Facility Deprecation
**How to categorize this issue?**

/area control-plane
/kind api-change

/platform equinix-metal

**What would you like to be added**:

The provider should more completely adapt to Metro based provisioning and remove Facility deployment options.

**Why is this needed**:

Equinix Metal is deprecating facility deployment. Metros encompass multiple facilities and are a better fit for availability concerns.

Metro provisioning is largely implemented in the same way, whether for IP addresses, VLANs, or Metal servers. The `metro` field is supplied *instead of* the `facility` (or `facilities`) field. The response object will include a `metro` rather than (in some cases, in addition to) a `facility` field. Network resources previously configured in a facility are available within the metro that facility resides within. All previously 'facility' provisioned network resources are portable across the metro with respect to the Metal Project (which has always been a secondary context for network resources).

https://feedback.equinixmetal.com/changelog/bye-facilities-hello-again-metros

Code of interest:
- https://github.com/search?q=repo%3Agardener%2Fgardener-extension-provider-equinix-metal%20ewr1&type=code
- https://github.com/search?q=repo%3Agardener%2Fgardener-extension-provider-equinix-metal%20facilities&type=code

related to:
* https://github.com/gardener/gardener-extension-provider-equinix-metal/pull/161
* https://github.com/gardener/machine-controller-manager-provider-equinix-metal/issues/16
* https://github.com/packethost/packngo/issues/369

### PR DESCRIPTION
**Pull request by schrodit**
**How to categorize this PR?**

/area quality
/kind enhancement
/platform equinix-metal

**What this PR does / why we need it**:

Updates the equinix sdk to use the newer equinix one instead of the old packet one

Fixes https://github.com/gardener/gardener-extension-provider-equinix-metal/issues/265

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/webhook/controlplane/ensurer_test.go [hunk 1]
```diff
@@ -150,94 +162,6 @@ var _ = Describe("Ensurer", func() {
 			Expect(err).To(Not(HaveOccurred()))
 			checkKubeAPIServerDeployment(dep, "1.26.0", annotations)
 		})
-
-		It("should keep the NODE_NETWORK env variable in the kube-apiserver deployment if its value does not change", func() {
```
Comment (schrodit) [inline pkg/webhook/controlplane/ensurer_test.go:154]: the vpn-seed conatiner in the apiserver does not exist anymore

**Review by RiRa12621 — APPROVED** [review]
/lgtm

maybe add a test but not blocking on that.

**Pull request merged**